### PR TITLE
Combos creation

### PR DESCRIPTION
**Pull request by Arnau77**
Creation of the different combos for the different worlds and define every final combo for every world with its collision box. Create the system to use skills (without cooldowns). Add the status effects' system. Add ignited, weakened and slowed status effects.

### CODE REVIEW

**Comment by daskza19** [comment]
I think there's a problem with my branch and yours. It is not normal that all my changes are inside your pr too. Viewing your commits i see that ([this](https://github.com/Pletenica/SunflowerProject/pull/12/commits/31dc346051fbd9e034980ef98b094fb1f492a01d)) has all changes. Can we see in the future the way you manage conflicts? It's dificult to me see this PR with all changes from two features... (113 files)

File: Sunflower Project/Assets/Scripts/Character/ST_Entity.cs [hunk 1]
```diff
@@ -39,6 +53,17 @@ public enum TypeOfDamage
...
 
+    //TODO: THIS WILL PROBABLY CHANGE IN THE FUTURE AND BECOME A LIST OF STATUS EFFECTS
+    /// <summary>
+    /// This defines if the entity has a status effect and which one it has
+    /// </summary>    
+    public StatusEffect statusEffect { get; private set; } = StatusEffect.NONE;
+
+    protected void Awake()
+    {
+        statusEffectsInfo = Resources.Load<ST_StatusEffectsInfo>("SO_StatusEffectsInfo");
```
Comment (daskza19) [inline Sunflower Project/Assets/Scripts/Character/ST_Entity.cs:64]: I think we can separate the status effects with a new class that can be not monobehaviour. In this class we can store the actual status effect from the entity. Is sufficient independ from the rest of logic of the entity.

File: Sunflower Project/Assets/Scripts/Managers/ST_GameManager.cs [hunk 2]
```diff
@@ -98,4 +132,13 @@ public void ChangePauseStateInGame(bool willBePaused)
...
         Time.timeScale = willBePaused ? 0 : 1;
     }
+
+    /// <summary>
+    /// This function is called when the player changes world
+    /// </summary>
+    /// <param name="newWorld"> This is the new world the player will change to</param>
+    public void ChangeWorld(World newWorld)
+    {
+        world = newWorld;
```
Comment (daskza19) [inline Sunflower Project/Assets/Scripts/Managers/ST_GameManager.cs:139]: I think we can create an Event that will be Invoke in this method. All the GO that will be affected by world changing can be suscribed in there.

**Review by daskza19 — CHANGES_REQUESTED** [review]

File: Sunflower Project/Assets/Scripts/Managers/ST_GameManager.cs (hunk 2 above)
Comment (Arnau77) [reply Sunflower Project/Assets/Scripts/Managers/ST_GameManager.cs:139]: This function will change in the future (when change of world is implemented), however, this comment is already as a TODO comment to take into account when changing the function.

**Review by daskza19 — APPROVED** [review]

**Pull request merged**